jdaviz/configs/imviz/plugins/links_control/links_control.vue

```diff
@@ -30,15 +41,87 @@
             persistent-hint>
           </v-switch>
         </v-row>
+        <div v-if="link_type_selected == 'WCS'">
+
+          <j-plugin-section-header>Select orientation</j-plugin-section-header>
+          <plugin-viewer-select
+            :items="viewer_items"
+            :selected.sync="viewer_selected"
+            :multiselect="multiselect"
+            :label="multiselect ? 'Viewers' : 'Viewer'"
+            :show_if_single_entry="multiselect"
+            :hint="multiselect ? 'Select viewers to set options simultaneously' : 'Select the viewer to set orientation'"
+          />
+          <plugin-layer-select
+            :items="layer_items"
+            :selected.sync="layer_selected"
+            :multiselect=false
+            :show_if_single_entry="true"
+            :label="'Orientation in viewer'"
+            :hint="'Select the viewer orientation'"
+          />
+          <v-row>
+            <span style="line-height: 36px">Presets:</span>
+            <j-tooltip tooltipcontent="Default orientation">
+              <v-btn icon @click="select_default_orientation">
+                <v-icon>mdi-restore</v-icon>
+              </v-btn>
+            </j-tooltip>
+            <j-tooltip tooltipcontent="north up, east left">
+              <v-btn icon @click="set_north_up_east_left">
+                <img :src="icon_nuel" width="24" class="invert-if-dark" style="opacity: 0.65"/>
+              </v-btn>
+            </j-tooltip>
+            <j-tooltip tooltipcontent="north up, east right">
+              <v-btn icon @click="set_north_up_east_right">
+                <img :src="icon_nuer" width="24" class="invert-if-dark" style="opacity: 0.65"/>
+              </v-btn>
+            </j-tooltip>
+          </v-row>
+        </div>
+
+        <div style="grid-area: 1/1">
+        </div>
+        <div v-if="link_type_selected == 'WCS'">
+
+          <j-plugin-section-header>Add orientation options</j-plugin-section-header>
+
+              <plugin-auto-label
+                :value.sync="new_layer_label"
+                :default="new_layer_label_default"
+                :auto.sync="new_layer_label_auto"
+                label="Name for orientation option"
+                hint="Label for this new model component."
+              ></plugin-auto-label>
+              <v-row>
+              <v-text-field
+                v-model="rotation_angle"
+                label="Rotation angle"
+                hint="Degrees counterclockwise from default orientation"
+                persistent-hint
+              ></v-text-field>

```

Comment:
this should have `type="number"` and use `IntHandleEmpty` or `FloatHandleEmpty` traitlet.

```suggestion
              <v-text-field
                v-model="rotation_angle"
                type="number"
                label="Rotation angle"
                hint="Degrees counterclockwise from default orientation"
                persistent-hint
                :rules="[() => roation_angle!=='' || 'This field is required']"
              ></v-text-field>

```

Reply:
I dropped the `type="number"` because negative numbers threw an error. Is there a way to fix that?

Reply:
you want to forbid negative rotations?  I would suggest doing that with more validation/rules so that they are caught before raising an error from python.

Reply:
The opposite – I want to allow negative numbers, but putting the non-numeric "-" character in the text field prevents the field from being interpreted with `type="number"`.

Reply:
I can't see this elsewhere where we use `type='number'` (line_profiles_xy in imviz, for example).  What is the error?

Reply:
I see this error in the log: 
```
TraitError: The 'rotation_angle' trait of a LinksControl instance expected a float, not the str '-30'.
```